File: C++/Euler7/euler7.cpp
#include <iostream>
#include <math.h>
int main(){
    //Clock Stuff
    std::clock_t start;
    double duration;
    start = std::clock();
    //Stop Clock Stuff
    
    int i=3,x=2;
    bool ifPrime=0;
    
    while (x<10001)
    {
        i+=2;
        for (int div=2;div<=sqrt(i);div++)
        {
            if (i%div==0) {ifPrime=0; break;}
            else ifPrime=1;
        }
        if (ifPrime==1) x++;
    }
    
    std::cout<<i<<std::endl;
    
    //std::cout << "The " << limit << " prime is:" << num << std::endl;
    
    duration = (std::clock() - start) / (double) CLOCKS_PER_SEC;
    std::cout<<"Elapsed Time: "<< duration << " seconds" << std::endl;
    return 0;
}

File: Fortran/euler7/euler7.f90
program euler7
implicit none
real :: start,finish

call cpu_time(start)
call ans()
call cpu_time(finish)

print'("Elapsed: ", F9.6, " seconds")', finish-start

contains
    subroutine ans()
    integer :: limit = 10001
    integer :: prime = 3
    integer :: num = 2
    integer :: i
    logical :: ifPrime = .false.

    do while (num < limit)
        prime = prime + 2
        do i = 2, floor(sqrt(real(prime)))
            if (mod(prime,i)==0) then
                ifPrime = .false.
                exit
            else
                ifPrime = .true.
            end if
        end do
        if (ifPrime .eqv. .true.) then
            num = num+1;
        end if
    end do

    print *, prime



    end subroutine ans
end program euler7